src/settings.rs

```diff
@@ -56,6 +58,17 @@ pub enum CharSize {
 	Bits8 = 8,
 }
 
+impl Display for CharSize {
+	fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+		match self {
+			CharSize::Bits5 => write!(f, "Five"),
+			CharSize::Bits6 => write!(f, "Six"),
+			CharSize::Bits7 => write!(f, "Seven"),
+			CharSize::Bits8 => write!(f, "Eight"),
+		}
```

Comment:
```suggestion
		Display::fmt(&self as u8, f)
```

Reply:
this had to be changed to ```Display::fmt(&(*self as u8), f)``` because I got a compilation error